src/api/stats/avax/getPangolinV2DualApys.js

```diff
@@ -0,0 +1,156 @@
+      .plus(yearlyRewardsBInUsd)
+      .plus(yearlyRewardsCInUsd);
+
+    const simpleApy = yearlyRewardsInUsd.dividedBy(totalStakedInUsd);
+    const vaultApr = simpleApy.times(shareAfterBeefyPerformanceFee);
+    const vaultApy = compound(simpleApy, BASE_HPY, 1, shareAfterBeefyPerformanceFee);
+
+    const tradingApr = tradingAprs[pool.address.toLowerCase()] ?? new BigNumber(0);
+    const totalApy = getFarmWithTradingFeesApy(
+      simpleApy,
+      tradingApr,
+      BASE_HPY,
+      1,
+      shareAfterBeefyPerformanceFee
+    );
+
+    // Create reference for legacy /apy
+    const legacyApyValue = { [pool.name]: totalApy };
+
+    apys = { ...apys, ...legacyApyValue };
+
+    // Create reference for breakdown /apy
+    const componentValues = {
+      [pool.name]: {
+        vaultApr: vaultApr.toNumber(),
+        compoundingsPerYear: BASE_HPY,
+        beefyPerformanceFee: beefyPerformanceFee,
+        vaultApy: vaultApy,
+        lpFee: liquidityProviderFee,
+        tradingApr: tradingApr.toNumber(),
+        totalApy: totalApy,
+      },
+    };
+
+    apyBreakdowns = { ...apyBreakdowns, ...componentValues };
+  }
+
+  // Return both objects for later parsing
+  return {
+    apys,
+    apyBreakdowns,
+  };
+};
+
+const getMasterChefData = async () => {
+  const masterchefContract = new web3.eth.Contract(MasterChef, masterchef);
+  const rewardPerSecond = new BigNumber(await masterchefContract.methods.rewardPerSecond().call());
+  const totalAllocPoint = new BigNumber(await masterchefContract.methods.totalAllocPoint().call());
+  return { rewardPerSecond, totalAllocPoint };
+};
+
+const getPoolsData = async pools => {
+  const masterchefContract = new web3.eth.Contract(MasterChef, masterchef);
+  const multicall = new MultiCall(web3, multicallAddress(AVAX_CHAIN_ID));
+  const balanceCalls = [];
+  const poolInfoCalls = [];
+  const rewarderCalls = [];
+  const multipliersCalls = [];
+  pools.forEach(pool => {
+    const tokenContract = new web3.eth.Contract(ERC20, pool.address);
+    balanceCalls.push({
+      balance: tokenContract.methods.balanceOf(masterchef),
+    });
+    let poolInfo = masterchefContract.methods.poolInfo(pool.poolId);
+    poolInfoCalls.push({
+      poolInfo: poolInfo,
+    });
+    let rewarder = masterchefContract.methods.rewarder(pool.poolId);
+    rewarderCalls.push({
+      rewarder: rewarder,
+    });
+  });
+
+  const res = await multicall.all([balanceCalls, poolInfoCalls, rewarderCalls]);
+
+  const balances = res[0].map(v => new BigNumber(v.balance));
+  const allocPoints = res[1].map(v => v.poolInfo[2]);
+  const rewarders = res[2].map(v => v.rewarder);
+
+  rewarders.forEach(rewarder => {
+    let rewarderContract = new web3.eth.Contract(Rewarder, rewarder);
+    let multiplier = rewarderContract.methods.getRewardMultipliers();
+    multipliersCalls.push({
+      multiplier: multiplier,
+    });
+  });
+
+  const multipliers = await multicall.all([multipliersCalls]);
```

Comment:
```suggestion
  const multipliers = (await multicall.all([multipliersCalls]))[0].map(m =>
    m.multiplier.map(v => new BigNumber(v))
  );
```

Reply:
Thanks for the thorough feedback. I relocated the mapping as you suggested